python/geom.py

```diff
@@ -541,7 +541,7 @@ def __init__(self,
                  medium2,
                  weights=None,
                  grid_type="U_DEFAULT",
-                 do_averaging=False,
+                 do_averaging=True,
```

Comment:
Yes this is intentional.

I can add an additional test.

Reply:
I have seen a significant performance regression when `do_averaging=True`, so it may be preferred to keep this off by default.

Reply:
You've seen a regression with this PR applied? Can you be more specific?

Reply:
The time taken to calculate the gradient is significantly higher with `do_averaging=True` than with `do_averaging=False`.

Reply:
Yes, on `master`. The point of this PR is to alleviate that issue. Have you applied this PR and seen similar regression?

Reply:
I have not.

Reply:
Just for reference, my own benchmarks (using both the meep tests and my own test files) show similar computational speeds when smoothing is off or on.

I'm sure you have many more tests you use to debug new features. If you want to run this on those, that would be great.

But as stated in #1903, the primary reason for disabling smoothing was because of the speed degradation. Since this PR supposedly resolves that, let's revert the default to true.

While I'm on my soapbox, I'll point out that the point of smoothing isn't so much to gain second-order acuraccy (the benefits of that seem to be relatively minor). Rather, it allows us to compute gradients at *any* beta (that's not true when smoothing is off, or for any traditional density method for that matter). So in short, enabling smoothing is very important from a practical design perspective.